### PR TITLE
OH2-276, OH2-277, OH2-278, OH2-279, OH2-280 | Implement admin[wards] feature

### PR DESCRIPTION
**Pull request by SteveGT96**
See [OH2-276](https://openhospital.atlassian.net/browse/OH2-276), [OH2-277](https://openhospital.atlassian.net/browse/OH2-277), [OH2-278](https://openhospital.atlassian.net/browse/OH2-278), [OH2-279](https://openhospital.atlassian.net/browse/OH2-279), [OH2-280](https://openhospital.atlassian.net/browse/OH2-280)

[OH2-276]: https://openhospital.atlassian.net/browse/OH2-276?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OH2-277]: https://openhospital.atlassian.net/browse/OH2-277?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OH2-278]: https://openhospital.atlassian.net/browse/OH2-278?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OH2-279]: https://openhospital.atlassian.net/browse/OH2-279?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OH2-280]: https://openhospital.atlassian.net/browse/OH2-280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by gasp** [comment]
on large resolution, there is visual glitch making the content space much smaller.
This should be a fixed width to prevent from disturbing the user flow

Screen Recording 2024-05-03 at 15 48 01

**Comment by gasp** [comment]
There is no e2e testing to support this feature. Should we consider adding it here or should we treat it separately?

File: src/types.ts [hunk 1]
```diff
@@ -223,4 +223,5 @@ export type TPermission =
   | "dashboard.access"
   | "laboratories.access"
   | "patients.access"
-  | "opds.access";
+  | "opds.access"
+  | "admin.access";
```
Comment (gasp) [inline src/types.ts:227]: good one 👍🏼

**Review by gasp — COMMENTED** [review]
Some minor comments, I suggest that we merge this so that it doesn't block other features

File: src/state/ward/actions.ts [hunk 2]
```diff
@@ -1,16 +1,32 @@
 import { isEmpty } from "lodash";
 import { Dispatch } from "redux";
-import { DiseaseDTO } from "../../generated";
+import { DiseaseDTO, WardDTO } from "../../generated";
```
Comment (gasp) [inline src/state/ward/actions.ts]:
```suggestion
import { WardDTO } from "../../generated";
```

This is unused and can be safely removed

File: src/types.ts (hunk 1 above)
Comment (gasp) [inline src/types.ts:227]: please keep in sync with `/mockServer/fixtures/permissionList.js` as described in the comment or deprecate the comment

File: src/routes/Admin/AdminRoutes.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
+import React, { ReactNode, useMemo } from "react";
+import { useTranslation } from "react-i18next";
+import { Navigate, Route, Routes } from "react-router";
+import NotFound from "../../components/activities/notFound/NotFound";
+import AdminActivity from "../../components/activities/adminActivity/AdminActivity";
+import { AdminActivityContent } from "../../components/activities/adminActivity/AdminActivityContent/AdminActivityContent";
+import { Wards } from "../../components/accessories/admin";
+import { NewWard } from "../../components/accessories/admin/wards/newWard/NewWard";
+import { EditWard } from "../../components/accessories/admin/wards/editWard/EditWard";
```
Comment (gasp) [inline src/routes/Admin/AdminRoutes.tsx]: maybe a shorter import could be more elegant

```suggestion
import { Wards, NewWard, EditWard } from "../../components/accessories/admin";
```
or 
```suggestion
import { Wards, NewWard, EditWard } from "../../components/accessories/admin/wards";
```

**Comment by SteveGT96** [comment]
> There is no e2e testing to support this feature. Should we consider adding it here or should we treat it separately?

Thanks @gasp for your review.

I think we'll better handle this in a separate issue.

File: src/components/accessories/admin/wards/wardTable/WardTable.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+          case "SUCCESS":
+            return (
+              <>
+                <Table
+                  rowData={formatDataToDisplay(data ?? [])}
+                  tableHeader={header}
+                  labelData={label}
+                  columnsOrder={order}
+                  rowsPerPage={2}
```
Comment (gasp) [inline src/components/accessories/admin/wards/wardTable/WardTable.tsx]:
```suggestion
                  rowsPerPage={10}
```
or maybe 20?

File: src/components/accessories/admin/wards/wardForm/index copy.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export * from "./WardForm";
```
Comment (gasp) [inline src/components/accessories/admin/wards/wardForm/index copy.ts]: file "index copy.ts" should not be aded to repository

**Review by gasp — APPROVED** [review]

**Review by mwithi — APPROVED** [review]
Approved by @gasp

**Comment by axelpezzo** [comment]
Hello @SteveGT96, below my review:

- Inside the Add / Edit form the "reset" button should be renamed "cancel", and his behaviour should be changed too because now once you enter on add / edit mode, there is no way to get back to the entries.
- On the top of the entries table the button that expand / compress the table should be align to the left and the "add ward" should be aligned to the right of the table.
- There are some style differences between the patient info vs hospital info, please check the font size of that.

**Comment by gasp** [comment]
@axelpezzo good catch. Can those be added as jira tickets alongside adding e2e testing?

Can we get rid of the "expand" behaviour as it serves little purpose and uses some precious real-estate?

**Comment by SteveGT96** [comment]
@gasp I've handled these comments in [#582](https://github.com/informatici/openhospital-ui/pull/582)

**Pull request merged**